### PR TITLE
fix(augmentation): share sampled motion blur kernel size across batch

### LINKED ISSUE #4671: RandomMotionBlur draws kernel_size per sample but blurs the whole batch with one of them
### Describe the bug

`RandomMotionBlur` draws `kernel_size` **once per sample** into `_params["ksize_factor"]`, and then blurs the **whole batch** with the single entry at `_params["idx"]`, an index drawn uniformly over the batch. Every other drawn size is discarded.

So a batch of 8 with `kernel_size=(3, 9)` does not get 8 independently blurred samples; it gets one kernel size applied to all 8, and the other 7 draws are dead. The stored `ksize_factor` has shape `(B,)` and reads as if it were per sample, which is what makes this easy to miss.

This is the *other half* of #4599. That issue was scoped to the truncated draw — the upper bound of a tuple range was practically never reached — and is fixed by #4610. The one-draw-for-the-batch behaviour is untouched by that fix and was pinned separately as a wart in `tests/augmentation/test_conventions_intensity_ops.py`, under a `_4599` suffix that will name a closed issue once #4610 merges. This issue exists to carry that pin.

### Reproduction steps

```python
import torch
import kornia.augmentation as K

torch.manual_seed(0)
aug = K.RandomMotionBlur((3, 9), (0.0, 0.0), (0.0, 0.0), p=1.0)
out = aug(torch.rand(8, 1, 16, 16))

print(aug._params["ksize_factor"])  # 8 different sizes were drawn
print(aug._params["idx"])           # only this one is used, for all 8 samples
```

Every sample in `out` is blurred with `ksize_factor[idx]`.

### Expected behavior

Either the kernel size is applied per sample, matching the per-sample shape of `ksize_factor` and the convention every other intensity augmentation follows; or the draw is made once and `ksize_factor` stops pretending to be per sample.

The per-sample route is the more expensive one — `kornia.filters.motion_blur` takes a single `kernel_size`, so a genuinely per-sample size needs the batch grouped by drawn size, or padded to a common kernel. That cost is presumably why it works this way, in which case the answer may be to document it rather than change it, and to make the parameter shape honest.

Either way the current state — a per-sample parameter that is silently reduced to one entry — is worth a decision.

### Environment

Measured on torch 2.5.0, cpu, on `main` at `e1b79c21` and on the `#4610` merge tree; the behaviour is identical on both, and `same_on_batch` does not affect it.

### Additional context

Raised out of review on #4610. `RandomMotionBlur3D` has a related but separate defect, filed separately.

### PR DESCRIPTION
**Pull request by Jeevang1-epic**
## What does this pull request do?

`RandomMotionBlur` currently samples a kernel size for every item in the batch, but only one of those sampled values is actually used to blur the whole batch.

That makes `ksize_factor` look per-sample even though the operation is batch-wide, and the remaining sampled kernel sizes are effectively discarded.

This change makes the kernel-size sampling match the actual behavior: ranged kernel sizes are sampled once for the batch and repeated in the existing parameter tensor.

Angles and directions keep their current per-sample behavior, and replay with previously supplied parameters still works through the existing indexed path.

I kept this scoped to the 2D `RandomMotionBlur` path and did not change the filtering implementation or the 3D augmentation.

## Related issue or discussion

Fixes #4671

## What did you check?

- Reproduced the current mismatch on unchanged `main`
- Added regression coverage for the sampled kernel size matching the kernel actually applied
- Checked both float32 and float64
- Checked fixed and ranged kernel sizes
- Checked `same_on_batch=True` and `same_on_batch=False`
- Checked seeded/replay behavior and existing parameter compatibility
- Checked the relevant `VideoSequential` replay paths
- Checked the existing 2D gradient test
- Ran the focused RandomMotionBlur and generator tests
- Ran the broader augmentation/convention/container test set
- Ran type checking
- Ran the full pre-commit hooks
- Ran final diff checks

Focused tests:
- 149 passed
- 2 skipped
- 3 expected failures

Broader relevant suite:
- 3928 passed
- 2580 skipped
- 6 expected failures

The new regression checks also fail against the unchanged upstream implementation, confirming they cover the behavior fixed here.

## How did AI help?

AI helped with repository inspection, refining the implementation, and a small part of the validation workflow. I reviewed the final changes and ran the local tests and Git checks before opening the PR.

### CODE REVIEW

**Comment by Jeevang1-epic** [comment]
hi @ducha-aiki   all checks are green now whenever you get a chance can you take a look at #4687

**Review by ducha-aiki — APPROVED** [review]
The generator now shares the ranged kernel-size draw across the batch as documented, while preserving per-sample angle/direction behavior. Focused motion-blur validation passed (26 tests). No blocking correctness findings.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).
